File: ruby/sieve/sieve.rb
class Sieve
  def initialize(max)
    @numbers = (1..max)
  end

  def primes
    @numbers.select { |i| prime?(i) }
  end

  private

    def prime?(n)
      (1..n).count { |i| n % i == 0 } == 2
    end
end

module BookKeeping
  VERSION = 1
end

File: kotlin/sieve/src/main/kotlin/Sieve.kt
import kotlin.math.sqrt

object Sieve {
    fun primesUpTo(max: Int): List<Int> {
        var numbers = (2..max).toList()
        val primes = mutableListOf<Int>()

        while (numbers.isNotEmpty()) {
            if (numbers.first() > sqrt(max.toDouble())) {
                primes.addAll(numbers)
                break
            }
            primes.add(numbers.first())
            numbers = numbers.filter { it % numbers.first() != 0 }
        }

        return primes
    }
}